ads/google/imaVideo.js

```diff
@@ -378,12 +384,37 @@ export function imaVideo(global, data) {
           false);
     });
 
+    // Handle settings that need to be set before the AdDisplayContainer is
+    // created.
+    if (imaSettings.locale) {
+      global.google.ima.settings.setLocale(imaSettings.locale);
+    }
+    if (imaSettings.vpaidMode) {
+      global.google.ima.settings.setVpaidMode(imaSettings.vpaidMode);
+    }
+
+
     adDisplayContainer =
         new global.google.ima.AdDisplayContainer(adContainerDiv, videoPlayer);
 
     adsLoader = new global.google.ima.AdsLoader(adDisplayContainer);
     adsLoader.getSettings().setPlayerType('amp-ima');
     adsLoader.getSettings().setPlayerVersion('0.1');
+    // Propogate settings provided via child script tag.
+    // locale and vpaidMode are set above, as they must be set before we create
+    // an AdDisplayContainer.
+    // playerType and playerVersion are used by the developers to track usage,
+    // so we do not want to allow users to overwrite those values.
+    const skippedSettings =
+        ['locale', 'vpaidMode', 'playerType', 'playerVersion'];
+    for (const setting in imaSettings) {
+      if (!skippedSettings.includes(setting)) {
+        // Change e.g. 'ppid' to 'setPpid'.
+        const methodName =
+            'set' + setting.charAt(0).toUpperCase() + setting.slice(1);
+        adsLoader.getSettings()[methodName](imaSettings[setting]);
```

Comment:
add safety check to prevent exception if user-provided setting happens to be wrong.